docs/csharp/tutorials/inheritance.md

```diff
@@ -0,0 +1,294 @@
+   {
+   }
+   ```
+
+## Implicit inheritance ##
+
+Besides any types that they may inherit from through single inheritance, all types in the .NET type system implicitly inherit from a base type. This ensures that common functionality is available to any type.
+
+To see what implicit inheritance means, let's define a new class, `SimpleClass`, that is simply an empty class definition:
+
+[!CODE [Inheritance](../../../samples/snippets/csharp/tutorials/inheritance/simpleclass.cs#1)]
+
+We can then use reflection (which lets us inspect a type's metadata to get information about that type) to get a list of the members that belong to the `SimpleClass` type. Although we haven't defined any members in our `SimpleClass` class, output from the example indicates that it actually has seven members. One of these is a parameterless (or default) constructor that is automatically supplied by the C# compiler. The other six are members of @System.Object, the type from which all classes and interfaces in the .NET type system ultimately implicitly inherit.
+
+[!CODE [Inheritance](../../../samples/snippets/csharp/tutorials/inheritance/simpleclass.cs#2)]
+
+Implicit inheritance from the @System.Object class makes these methods available to the `SimpleClass` class:
+
+- The public `ToString` method, which converts a `SimpleClass` object to its string representation, the fully qualified type name. In this case, the `ToString` method returns the string "SimpleClass".
+
+- The public `Equals` method, which tests two objects to determine whether they are equal. The method tests for reference equality; that is, to be equal, two object variables must refer to the same object.
+
+- The public `GetHashCode` method, which computes a value that allows an instance of the type to be used in hashed collections.
+
+- The public `GetType` method, which returns a @System.Type object that represents the `SimpleClass` type.
+
+- The protected @System.Object.Finalize method, which is designed to release unmanaged resources before an object's memory is reclaimed by the garbage collector.
+
+- The protected @System.Object.MemberwiseClone method, which creates a shallow clone of the current object.
+
+Because of implicit inheritance, we can call any inherited member from a `SimpleClass` object just as if it was actually a member defined in the `SimpleClass` class. For instance, the following example calls the `SimpleClass.ToString` method, which `SimpleClass` inherits from @System.Object.
+
+[!CODE [Inheritance](../../../samples/snippets/csharp/tutorials/inheritance/simpleclass2.cs#1)]
+
+The following table lists the categories of types that you can create in C# and the types from which they implicitly inherit. Each base type makes a different set of members available through inheritance to implicitly derived types.
+
+| Type category | Implicitly inherits from... |
+| :--- | :---: | ---: |
+| class | @System.Object |
+| struct | @System.ValueType |
+| interface | @System.Object |
```

Comment:
I would delete this.  An interface definition does not implicitly inherit from System.Object. (If it did, any class that implemented interfaces would run into the issues of multiple inheritance in C++ where multiple copies of a base class data and methods can appear. Bad things).

Every type that implements an interface will inherit from System.Object though. For that reason, it may be worth stating that System.ValueType derives from System.Object. System.MulticastDelegate also derives from System.Object.